libevmasm/JumpdestRemover.cpp

```diff
@@ -59,11 +59,27 @@ set<size_t> JumpdestRemover::referencedTags(AssemblyItems const& _items, size_t
 {
 	set<size_t> ret;
 	for (auto const& item: _items)
-		if (item.type() == PushTag)
+	{
+		switch (item.type())
+		{
+		case PushTag:
 		{
 			auto subAndTag = item.splitForeignPushTag();
 			if (subAndTag.first == _subId)
 				ret.insert(subAndTag.second);
+			break;
+		}
+		case JumpTablePushTag:
+		{
+			// TODO: Do we need to look at the sub here?
```

Comment:
Jump table tags should always refer to the current sub-assembly, shouldn't they? We may want to assert this here, but don't need more than that.

Reply:
That's what I was thinking, but I wasn't exactly sure of the sub-assembly architecture. Thank you for confirming that. I don't store the sub in the JumpTablePushTag anyway, so I just removed the comment.